hwcomposer/hwcomposer_backend_v11.cpp

```diff
@@ -346,9 +586,63 @@ HwComposerBackend_v11::swap(EGLNativeDisplayType display, EGLSurface surface)
 #endif
 }
 
+int
+HwComposerBackend_v11::present(RetireFencePool *pool, buffer_handle_t handle, int acquireFenceFd)
+{
+    bool geometryChanged = m_screenAttachedGeometryChanged;
+    m_screenAttachedGeometryChanged = false;
+
+    content->prepare(handle, acquireFenceFd, width, height, geometryChanged);
+
+    auto display_list = content->get();
+
+    // Collect and reset all retire fence fds
+    for (int i=0; i<num_displays; i++) {
+        pool->consume(display_list[i]->retireFenceFd);
+    }
+
+    QPA_HWC_TIMING_SAMPLE(syncTime);
+
+    HWC_PLUGIN_EXPECT_ZERO(hwc_device->prepare(hwc_device, num_displays, display_list));
+
+    QPA_HWC_TIMING_SAMPLE(prepareTime);
+
+    QSystrace::begin("graphics", "QPA::set", "");
+    HWC_PLUGIN_EXPECT_ZERO(hwc_device->set(hwc_device, num_displays, display_list));
+    QSystrace::end("graphics", "QPA::set", "");
+
+    QPA_HWC_TIMING_SAMPLE(setTime);
+
+    return content->release();
+}
+
+void HwComposerBackend_v11::blankDisplay(int display, bool blank)
+{
+    int status;
+#ifdef HWC_DEVICE_API_VERSION_1_4
+    if (hwc_version == HWC_DEVICE_API_VERSION_1_4) {
+        status = hwc_device->setPowerMode(hwc_device, display, blank ? HWC_POWER_MODE_OFF : HWC_POWER_MODE_NORMAL);
+    } else
+#endif
+#ifdef HWC_DEVICE_API_VERSION_1_5
+    if (hwc_version == HWC_DEVICE_API_VERSION_1_5) {
+        status = hwc_device->setPowerMode(hwc_device, display, blank ? HWC_POWER_MODE_OFF : HWC_POWER_MODE_NORMAL);
+    } else
+#endif
+        status = hwc_device->blank(hwc_device, display, blank);
+    HWC_PLUGIN_EXPECT_ZERO(status);
+    g_unblanked_displays[display] = (status == blank);
+
+    // on certian devices this is needed
+    screenPlugged();
+}
+
 void
 HwComposerBackend_v11::sleepDisplay(bool sleep)
 {
+    // XXX: For debugging only
+    //dump_attributes(hwc_device, num_displays);
```

Comment:
Ideally such debug traces would be put behind Qt logging sub-categories for enabling them without recompilation